openshift-kube-apiserver/admission/customresourcevalidation/imagecontentsourcepolicy/validate_imagecontentsourcepolicy_test.go

```diff
@@ -0,0 +1,95 @@
+package imagecontentsourcepolicy
+
+import (
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclientfake "github.com/openshift/client-go/config/clientset/versioned/fake"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateICSPUse(t *testing.T) {
+
+	for _, tt := range []struct {
+		name           string
+		objects        []runtime.Object
+		action         string
+		validateErrors func(t *testing.T, errs field.ErrorList)
+	}{
+		{
+			name: "can't create icsp with existing idms",
+			objects: []runtime.Object{
+				&configv1.ImageDigestMirrorSet{
+					ObjectMeta: metav1.ObjectMeta{Name: "idms"},
+					Spec: configv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []configv1.ImageDigestMirrors{
+							{Source: "example.com", Mirrors: []configv1.ImageMirror{"mirror.com"}},
+						},
+					},
+				},
+			},
+			action: "create",
+
+			validateErrors: func(t *testing.T, errs field.ErrorList) {
+				t.Helper()
+				if len(errs) != 1 {
+					t.Fatal(errs)
+				}
+				if errs[0].Error() != `Kind.ImageContentSourcePolicy: Forbidden: can't create ImageContentSourcePolicy when ImageDigestMirrorSet resources exist` {
+					t.Error(errs[0])
+				}
+			},
+		},
+		{
+			name: "can't create icsp with existing itms",
+			objects: []runtime.Object{
+				&configv1.ImageTagMirrorSet{
+					ObjectMeta: metav1.ObjectMeta{Name: "itms"},
+					Spec: configv1.ImageTagMirrorSetSpec{
+						ImageTagMirrors: []configv1.ImageTagMirrors{
+							{Source: "example.com", Mirrors: []configv1.ImageMirror{"mirror.com"}},
+						},
+					},
+				},
+			},
+			action: "create",
+
+			validateErrors: func(t *testing.T, errs field.ErrorList) {
+				t.Helper()
+				if len(errs) != 1 {
+					t.Fatal(errs)
+				}
+				if errs[0].Error() != `Kind.ImageContentSourcePolicy: Forbidden: can't create ImageContentSourcePolicy when ImageTagMirrorSet resources exist` {
+					t.Error(errs[0])
+				}
+			},
+		},
+		{
+			name:   "success creating icsp",
+			action: "create",
+
+			validateErrors: func(t *testing.T, errs field.ErrorList) {
+				t.Helper()
+				if len(errs) > 0 {
+					t.Fatal(errs)
+				}
+			},
+		},
```

Comment:
You're missing test cases for update and updatestatus actions.

Reply:
The imagecontentsourcepolicy does not have a status field. Add the test for update action.